pytorch_lightning/plugins/precision/deepspeed_precision.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, Union
+
+import torch
+from torch.optim import Optimizer
+
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins.precision.precision_plugin import PrecisionPlugin
+
+
+class DeepSpeedPrecisionPlugin(PrecisionPlugin):
+
+    def __init__(self, precision):
+        super().__init__()
+        self.precision = precision
```

Comment:
@carmocca actually not, since the base plugin is default precision (aka fp32), so there is not such option.

Reply:
I see.

Should this hardcode `precision=16` then? Doesn't seem like deepspeed supports any other
https://www.deepspeed.ai/features/#distributed-training-with-mixed-precision

Reply:
Yes, that's probably the best then :)

Reply:
It technically supports float32, which is just basically turn off the precision plugin, but this 'precision plugin' still needs to be set.

Reply:
To recap for any future readers:

This isnt really a precision plugin, however, it is used because the hooks provided by the precision plugins are necessary for the deepspeed implementation.

Since this piece has nothing to do with precision, both 16 and 32 are allowed via the constructor.